code/controllers/subsystem/air.dm

```diff
@@ -353,6 +368,20 @@ SUBSYSTEM_DEF(air)
 		if(MC_TICK_CHECK)
 			return
 
+/datum/controller/subsystem/air/proc/process_hot_groups(resumed = FALSE)
+	if (!resumed)
+		src.currentrun = hot_groups.Copy()
+	var/list/currentrun = src.currentrun
+	while(currentrun.len)
+		var/datum/hot_group/HG = currentrun[currentrun.len]
+		currentrun.len--
+		if (HG)
+			HG.process()
+		else
+			hot_groups -= HG
+		if(MC_TICK_CHECK)
+			return
+
```

Comment:
I am still not sure how I feel about giving this first class support. My gut is merger datums but they do not provide the flexability you need to make this reasonable/fast, so idk. maybe this is fine